### PR TITLE
Add Medications section

### PR DESCRIPTION
**Pull request by dsamojlenko**
## Adds
This PR re-introduces the main Medications section workflow including:
- Medications list screen
- Add Medication
- Edit Medication
- Delete Medication

## Excluded
This PR *does not* include:
- Some of the conditional validation may be incomplete.
- Does not redirect to Treatments at the end (goes back to Conditions for now)

## Locales
- The locales files keep getting regenerated - I haven't dug in to find out if it's due to missing strings, or if there's just something weird going on ... I recommend ignoring the locale files until we can do a sweep later.

## To test
- fill out the Personal information form
- Add one or more conditions
- Proceed to Medications
- Add one or more Medications
- Edit a Medication
- Delete a Medication

### CODE REVIEW

File: api/controllers/AddMedicationController.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  create: function (req, res) {
+    const data = req.session.medicalReport
+    const conditionList = conditionReducer(data.conditions)
+
+    res.view('pages/medications/add', {
+      conditionList: conditionList
+    });
+  },
+
+  store: function (req, res) {
```
Comment (jeffmaher) [inline api/controllers/AddMedicationController.js]: Requested change: I think LGTM is also reporting this, but it's having trouble with this line and the app won't start.

```
error: Attempted to `require('/Users/jeff/code/cppd-medical-report-sails/api/controllers/AddMedicationController.js')`, but an error occurred:
--
/Users/jeff/code/cppd-medical-report-sails/api/controllers/AddMedicationController.js:13
  store: function (req, res) {
  ^^^^^

SyntaxError: Unexpected identifier
    at Module._compile (internal/modules/cjs/loader.js:895:18)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:995:10)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Function.Module._load (internal/modules/cjs/loader.js:727:14)
    at Module.require (internal/modules/cjs/loader.js:852:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/include-all/lib/help-include-all-sync.js:293:33
    at Array.forEach (<anonymous>)
    at _recursivelyIncludeAll (/Users/jeff/code/cppd-medical-report-sails/node_modules/include-all/lib/help-include-all-sync.js:178:11)
    at includeAll (/Users/jeff/code/cppd-medical-report-sails/node_modules/include-all/lib/help-include-all-sync.js:317:5)
    at helpBuildDictionary (/Users/jeff/code/cppd-medical-report-sails/node_modules/include-all/lib/help-build-dictionary.js:46:13)
    at Function.module.exports.optional (/Users/jeff/code/cppd-medical-report-sails/node_modules/include-all/index.js:67:10)
    at loadActionModules (/Users/jeff/code/cppd-medical-report-sails/node_modules/sails/lib/app/private/controller/load-action-modules.js:27:14)
    at async.auto.controller (/Users/jeff/code/cppd-medical-report-sails/node_modules/sails/lib/app/load.js:102:9)
    at runTask (/Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:1619:13)
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:1559:13
    at processQueue (/Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:1569:13)
    at taskComplete (/Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:1588:9)
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:1612:17
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:906:16
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/sails/lib/app/load.js:192:14
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:3861:9
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:421:16
    at replenish (/Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:941:25)
    at iterateeCallback (/Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:931:17)
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:906:16
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:3858:13
    at afterwards (/Users/jeff/code/cppd-medical-report-sails/node_modules/sails/lib/app/private/loadHooks.js:351:16)
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:3861:9
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:421:16
    at replenish (/Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:941:25)
    at iterateeCallback (/Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:931:17)
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:906:16
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:3858:13
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:421:16
    at iteratorCallback (/Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:998:13)
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/sails/node_modules/async/dist/async.js:906:16
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/sails/lib/app/private/loadHooks.js:241:38
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/async-listener/glue.js:188:31
    at processTicksAndRejections (internal/process/task_queues.js:75:11)
```

File: api/utils/condition.mapper.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/**
+ * I don't know if we should convert this to a "sails helper"
+ */
+// const reduce = require('lodash.reduce')
+
+const conditionReducer = (conditions) => {
```
Comment (jeffmaher) [inline api/utils/condition.mapper.js]: Suggestion: Add a quick bit of code comments / documentation to this file to explain what the structure of the `conditions` parameter looks like when it is passed in and what something looks like with it leaves `conditionReducer`.

File: api/controllers/AddMedicationController.js (hunk 1 above)
Comment (dsamojlenko) [reply api/controllers/AddMedicationController.js]: hmm... I don't see this at all? Like the code at the line number quoted in the error message is different even? Maybe it got fixed when updating from master?

File: api/utils/condition.mapper.js (hunk 2 above)
Comment (dsamojlenko) [reply api/utils/condition.mapper.js]: Added some comments. This was a bit of code copied straight over from previous - think it's Calvin's ... think I got the gist of what it's doing. May be worth revisiting this later.

**Pull request merged**